pkg/cloud/api/path.go

```diff
@@ -29,20 +29,32 @@ import (
 // - "!" is a slice index
 // - ":" is a map key
 // - "*" is a pointer deref.
+// - "#" is all array elements reference
```

Comment:
I think we should have this in a different comment block as this character is not used as a prefix:

```
const (
  ...
  pathPointer = '#'

  // anyIndex is a wildcard index used instead of the number (slice) or string key (map).
  anyIndex = '#'
)
```

Reply:
Now that I think about it, isn't it much simpler to do this:

```
anySliceIndex = string(pathSliceIndex) + "#"
anyMapIndex = string(pathMapIndex) + "#"
```

then we don't need all the extra code below?